src/views/withNavigation.js

```diff
@@ -0,0 +1,19 @@
+/* @flow */
+
+import React from 'react';
+
+import { ContextWithNavigation } from '../TypeDefinition';
+
+export default function withNavigation(Component: ReactClass<T>) {
+  const componentWithNavigation = (props: T, { navigation }: ContextWithNavigation) => (
+    <Component {...props} navigation={navigation} />
+  );
+
+  componentWithNavigation.displayName = `withNavigation(${Component.displayName || Component.name})`;
+
+  componentWithNavigation.contextTypes = {
+    navigation: React.PropTypes.object.isRequired,
+  };
+
+  return componentWithNavigation;
```

Comment:
missing hoistStatics like this: https://github.com/exponent/ex-navigation/blob/79a7a5b87007bffe35b0947ac23996b20be9d6e1/src/ExNavigationComponents.js#L199